sdk/ai/Azure.AI.Agents.Persistent/samples/Sample19_PersistentAgents_VectorStoreFile_WithSteps.md

```diff
@@ -0,0 +1,283 @@
+# Sample file search using `VectorStoreFile` and agents in Azure.AI.Agents.Persistent.
+
+In this example we will create the local file, upload it and its file ID to new vector store using. Then we will drill down into run steps agent have ran through.
+1. First we need to create agent client and read the environment variables that will be used in the next steps.
+
+```C# Snippet:PersistentAgents_VectorStoreFileSearch_CreateClient
+var projectEndpoint = System.Environment.GetEnvironmentVariable("PROJECT_ENDPOINT");
+var modelDeploymentName = System.Environment.GetEnvironmentVariable("MODEL_DEPLOYMENT_NAME");
+PersistentAgentsClient client = new(projectEndpoint, new DefaultAzureCredential());
+```
+
+2. Create a file and upload it to the data store.
+
+Synchronous sample:
+```C# Snippet:PersistentAgents_VectorStoreFileSearch_FilesToUse_Sync
+// Upload a file and wait for it to be processed
+System.IO.File.WriteAllText(
+    path: "sample_file_for_upload.txt",
+    contents: "The word 'apple' uses the code 442345, while the word 'banana' uses the code 673457.");
+PersistentAgentFileInfo uploadedAgentFile = client.Files.UploadFile(
+    filePath: "sample_file_for_upload.txt",
+    purpose: PersistentAgentFilePurpose.Agents);
+Dictionary<string, string> fileIds = new()
+{
+    { uploadedAgentFile.Id, uploadedAgentFile.Filename }
+};
+```
+
+Asynchronous sample:
+```C# Snippet:PersistentAgents_VectorStoreFileSearch_FilesToUse_Async
+// Upload a file and wait for it to be processed
+System.IO.File.WriteAllText(
+    path: "sample_file_for_upload.txt",
+    contents: "The word 'apple' uses the code 442345, while the word 'banana' uses the code 673457.");
+PersistentAgentFileInfo uploadedAgentFile = await client.Files.UploadFileAsync(
+    filePath: "sample_file_for_upload.txt",
+    purpose: PersistentAgentFilePurpose.Agents);
+Dictionary<string, string> fileIds = new()
+{
+    { uploadedAgentFile.Id, uploadedAgentFile.Filename }
+};
+```
+
+3. To create agent capable of using file search, we will create `VectorStore` and add file ID to the it using `CreateVectorStoreFileAsync`.
```

Comment:
"To create an Agent"
", we will create a `VectorStore`"

Fix "to the it"

Reply:
Fixed